frontend/src/components/VExternalSearch/VExternalSearchForm.vue

```diff
@@ -133,7 +125,7 @@ export default defineComponent({
   },
   setup(_, { emit }) {
     const sectionRef = ref<HTMLElement | null>(null)
-    const triggerRef = ref<InstanceType<typeof VButton> | null>(null)
+    const triggerRef = ref<{ $el: HTMLElement } | null>(null)
```

Comment:
This is not strictly necessary for this PR, but it stops WebStorm from throwing an error in detecting the type of the triggerRef.

Reply:
I think that is because this file isn't included in `tsconfig.json`... if we're making changes to appease a particular IDE but ignoring the fact that VSCode, the most popular free IDE, shows heaps of errors on all the imports in these files that are marked `type="ts"` without being in the tsconfig, then I have a feeling we're not prioritising things correctly. Either we should remove the `type="ts"` from these files and not play whack-a-mole with errors like this, that actually _reduce_ type information and require manually writing types that do not need to be manually maintained or once and for all finally add everything to the tsconfig.

Critically, `type="ts"` on these files that are not included in the tsconfig are really a shadow pattern because it provides  false sense of security. It _looks_ like the file is type checked, but actually it never is. And on top of that it creates confusing issues for people setting up the project in anything that uses the normal `vue-tsc` rather than JetBrains's own implementation.

Anyway, I'm not suggesting to do anything in this PR. Just pointing out again that we're doing something very strange and I think actually quite risky (in the false-confidence sense of things) by working with TypeScript this way.

Reply:
My comment was actually misleading, sorry. The problem occurs both in WebStorm and in VS Code. Here's the message you get in both:
```
The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.ts(7056)

The inferred type of 'default' cannot be named without a reference to '.pnpm/@sentry+core@7.57.0/node_modules/@sentry/core'. This is likely not portable. A type annotation is necessary.ts(2742)
```
I suppose it's having the problem with the `default` imported from `VButton` that we use for annotation, and is replacing it with the `default` imported from `sentry`, for some reason. [This issue ](https://github.com/microsoft/TypeScript/issues/47663) says that it's a bug with TypeScript and pnpm. There is a workaround in the issue, but I couldn't understand how to get it to work.

On your point of files not being in `tsconfig`, I completely agree. However, we have recently added all of the `components/` to `tsconfig`:
https://github.com/WordPress/openverse/blob/dfd98eeced8d20f5f2684b0676a58f59a4d6435f/tsconfig.json#L40
We still need to add some pages (most notably `search.vue`)

Reply:
Gotcha, no worries then.

Do you know why this doesn't cause an issue when running the type check?

Reply:
> Do you know why this doesn't cause an issue when running the type check?

No, I don't. I tried adding that component to `tsconfig` manually, and type check still passed. When I also added `search.vue`, type check failed. So, I assume that type checks work, and all components **do** pass the type checks.